sematic/caching/git.py

```diff
@@ -0,0 +1,42 @@
+# Sematic
+from sematic.abstract_future import AbstractFuture
+from sematic.utils.git import get_git_info
+
+
+def get_git_sha_caching_namespace(root_future: AbstractFuture) -> str:
+    """
+    Returns a caching namespace based on the specified pipeline root Future func's source
+    code git commit SHA.
+
+    Must only be used when launching the Resolution from a development environment that is
+    versioned by git! Otherwise, this will raise an exception, and the cache will not be
+    used!
+
+    When using this caching namespace, switching to another commit or locally editing the
+    code would result in cache invalidation.
+
+    Successive code edits will not invalidate the cache, as the dirty code status is only
+    compared against the clean commit, and the actual contents of the code edits is not
+    considered.
+
+    Parameters
+    ----------
+    root_future: AbstractFuture
+        The root Future of the pipeline for which to look up the code git commit status.
```

Comment:
Nit: This function does not need to know/care that this future is a root future. This can be renamed to `future`.

Reply:
Done.